exporters/otlp/include/opentelemetry/exporters/otlp/recordable.h

```diff
@@ -21,7 +21,16 @@ class Recordable final : public sdk::trace::Recordable
   void SetAttribute(nostd::string_view key,
                     const opentelemetry::common::AttributeValue &value) noexcept override;
 
-  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept override;
+  void AddEvent(
+      nostd::string_view name,
+      core::SystemTimestamp timestamp = core::SystemTimestamp(std::chrono::system_clock::now()),
+      const trace::KeyValueIterable &attributes =
+          trace::KeyValueIterableView<std::map<std::string, int>>({})) noexcept override;
+
+  void AddLink(
+      opentelemetry::trace::SpanContext span_context,
+      const trace::KeyValueIterable &attributes =
+          trace::KeyValueIterableView<std::map<std::string, int>>({})) noexcept override;
```

Comment:
Substituting this at each call site might be expensive if most of the cases there is no attribute?

Reply:
Do you mean I should only define the default parameters in the base class?

Reply:
It will be a good optimization if we can avoid creating `std::map` if there is no attribute at all.

Reply:
Not blocking this PR though. Let me know if you think this is ready to get merged.

Reply:
I see. In this case what would the default parameter be? Or should we not use a default parameter?

Reply:
I think it's ready if there are no other comments!

Reply:
It could be a static object shared across the application (as long as we don't write to it), or some syntax sugar which wraps a nullptr.